adapters/ix/ix.go

```diff
@@ -359,3 +508,104 @@ func BuildIxDiag(request *openrtb2.BidRequest) error {
 	}
 	return nil
 }
+
+// moves sid from imp[].ext.bidder.sid to imp[].ext.sid
+func moveSid(imp *openrtb2.Imp) error {
+	if imp.Ext == nil || len(imp.Ext) == 0 {
+		return nil
+	}
+
+	var bidderExt adapters.ExtImpBidder
+	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		return err
+	}
+
+	var ixExt openrtb_ext.ExtImpIx
+	if err := json.Unmarshal(bidderExt.Bidder, &ixExt); err != nil {
+		return err
+	}
+
+	if ixExt.Sid != "" {
+		var m map[string]interface{}
+		if err := json.Unmarshal(imp.Ext, &m); err != nil {
+			return err
+		}
+		m["sid"] = ixExt.Sid
+		ext, err := json.Marshal(m)
+		if err != nil {
+			return err
+		}
+		imp.Ext = ext
+	}
+	return nil
+}
+
+func setFeatureToggles(a *IxAdapter, ext *json.RawMessage) {
+	if *ext == nil {
```

Comment:
Should this be `if ext == nil`? You want to verify the `ext` has a non-nil pointer before dereferencing. If this is indeed a mistake, consider adding test coverage to catch it.